src/Generators/staticgenerators.jl

```diff
@@ -0,0 +1,345 @@
+multipartite graph with `n` vertices and `r` partitions.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+- `r::Integer`: The number of partitions in the graph
+"""
+struct Turan{T<:Integer, U<:Integer} <: StaticGenerator
+    n::T
+    r::U
+    function Turan(n::T, r::U) where {T<:Integer, U<:Integer}
+        r in Base.OneTo(n) || throw(DomainError((n, r), "The number of partitions ($r) in the graph must not exceed the number of vertices ($n)."))
+        new{T, U}(n, r)
+    end
+end
+
+"""
+    struct Star <: StaticGenerator
+
+A struct representing a generator for a [star graph](https://en.wikipedia.org/wiki/Star_(graph_theory))
+with `n` vertices.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+"""
+struct Star{T<:Integer} <: StaticGenerator
+    n::T
+end
+
+"""
+    struct Path <: StaticGenerator
+
+A struct representing a generator for a [path graph](https://en.wikipedia.org/wiki/Path_graph)
+with `n` vertices.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+"""
+struct Path{T<:Integer} <: StaticGenerator
+    n::T
+end
+
+"""
+    struct Cycle <: StaticGenerator
+
+A struct representing a generator for a [cycle graph](https://en.wikipedia.org/wiki/Cycle_graph)
+with `n` vertices.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+"""
+struct Cycle{T<:Integer} <: StaticGenerator
+    n::T
+end
+
+"""
+    struct Wheel <: StaticGenerator
+
+A struct representing a generator for a [wheel graph](https://en.wikipedia.org/wiki/Wheel_graph)
+with `n` vertices.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+"""
+struct Wheel{T<:Integer} <: StaticGenerator
+    n::T
+end
+
+"""
+    struct Grid <: StaticGenerator 
+
+A struct representing a generator for a ``|dims|``-dimensional cubic lattice,
+with length `dims[i]` in dimension `i`.
+
+### Required Fields
+- `dims::AbstractVector{<:Integer}`: The lengths of the dimensions in the lattice. Note: this can also be a `Tuple`.
+
+### Optional Arguments
+- `periodic=false`: If true, the resulting lattice will have periodic boundary
+condition in each dimension.
+"""
+struct Grid{T<:Integer, U<:AbstractVector{T}} <: StaticGenerator
+    dims::U
+    periodic::Bool
+end
+Grid(dims; periodic=false) = Grid(dims, periodic)
+Grid(dims::Tuple; periodic=false) = Grid(collect(dims), periodic)
+
+"""
+    struct BinaryTree <: StaticGenerator
+
+A struct representing a generator for a complete [binary tree](https://en.wikipedia.org/wiki/Binary_tree)
+of depth `k`.
+
+### Required Fields
+- `k::Integer`: The depth of the binary tree
+"""
+struct BinaryTree{T<:Integer} <: StaticGenerator
+    k::T
+    function BinaryTree(k::T) where {T<:Integer}
+        if LightGraphs.isbounded(k) && BigInt(2) ^ k > typemax(k)
+            throw(OverflowError("k=$k will result in a graph too big for graph type $T"))
+        end
+        new{T}(k)
+    end
```

Comment:
I wonder if we should have this check here not in the concrete implementation.

Reply:
Each concrete implementation would have to implement it, which seems bad. Since AbstractGraphs will always be (at least) parameterized by the vertex index `<:Integer`, I think it's safe to do here.